examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,479 @@
+    print("Device:", tpu.master())
+    tf.config.experimental_connect_to_cluster(tpu)
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except:
+    strategy = tf.distribute.get_strategy()
+print("Number of replicas:", strategy.num_replicas_in_sync)
+
+"""
+We need a Google Cloud link to our data to load the data using a TPU. While we're at it,
+we instantiate constant variables. It is generally better practice to use constant
+variables instead of hard-coding numbers.
+"""
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+GCS_PATH = "gs://kds-7c9306925365b635aa934a70a0d94688c717d8c2eda0e47466736307"
+BATCH_SIZE = 16 * strategy.num_replicas_in_sync
+IMAGE_SIZE = [180, 180]
+
+"""
+## Load the data
+
+The Chest X-ray data we are using from
+[*Cell*](https://www.kaggle.com/paultimothymooney/chest-xray-pneumonia) divides the data
+into train, val, and test files. There are only 16 files in the validation folder, and we
+would prefer to have a less extreme division between the training and the validation set.
+We will append the validation files and create a new split that resembes the standard
+80:20 division instead.
+"""
+
+filenames = tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/train/*/*"))
+filenames.extend(tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/val/*/*")))
+
+filenames = np.array(filenames)
+np.random.shuffle(filenames)
+split_ind = int(0.8 * len(filenames))
+
+train_filenames, val_filenames = filenames[:split_ind], filenames[split_ind:]
+
+"""
+Run the following cell to see how many healthy/normal chest X-rays we have and how many
+pneumonia chest X-rays we have.
+"""
+
+COUNT_NORMAL = len([filename for filename in train_filenames if "NORMAL" in filename])
+print("Normal images count in training set: " + str(COUNT_NORMAL))
+
+COUNT_PNEUMONIA = len(
+    [filename for filename in train_filenames if "PNEUMONIA" in filename]
+)
+print("Pneumonia images count in training set: " + str(COUNT_PNEUMONIA))
+
+"""
+Notice that the there are way more images that are classified as pneumonia than normal.
+This shows that we have a imbalance in our data. We will correct for this imbalance later
+on in our notebook.
+"""
+
+train_list_ds = tf.data.Dataset.from_tensor_slices(train_filenames)
+val_list_ds = tf.data.Dataset.from_tensor_slices(val_filenames)
+
+for f in train_list_ds.take(5):
+    print(f.numpy())
+
+"""
+Run the following cell to see how many images we have in our training dataset and how
+many images we have in our validation set. Verify that the ratio of images is 80:20.
+"""
+
+TRAIN_IMG_COUNT = tf.data.experimental.cardinality(train_list_ds).numpy()
+print("Training images count: " + str(TRAIN_IMG_COUNT))
+
+VAL_IMG_COUNT = tf.data.experimental.cardinality(val_list_ds).numpy()
+print("Validating images count: " + str(VAL_IMG_COUNT))
+
+"""
+As expected, we have two labels for our images.
+"""
+
+CLASS_NAMES = np.array(
```

Comment:
Why make it an array?

In general avoid casting strings to `np.array` because the memory consumption is extremely high and you will OOM even for relatively small lists of strings. If you need a tensor-like object, prefer using `tf.constant`

In this case you can just leave it as list